packages/leancode_lint/lib/lints/missing_cleanup.dart

```diff
@@ -0,0 +1,374 @@
+    this.ignoredTypesCheckers = const [],
+    this.cleanupMethods = const {},
+  });
+
+  factory MissingCleanupConfig.fromConfig(Map<String, YamlNode?> json) {
+    final ignoredTypes =
+        (json['ignored_types'] as YamlList?)?.nodes
+            .map(
+              (e) => _IgnoredTypes(
+                name: (e as YamlMap)['ignore'] as String,
+                packageName: e['from_package'] as String,
+              ),
+            )
+            .toSet() ??
+        const {};
+
+    final cleanupMethodsMap = (json['cleanup_methods'] as YamlMap?)?.nodes.map(
+      (key, value) => MapEntry(
+        (key as YamlScalar).value as String,
+        (value as YamlScalar).value as bool,
+      ),
+    );
+
+    final cleanupMethods = {
+      if (cleanupMethodsMap?['dispose'] ?? true) 'dispose',
+      if (cleanupMethodsMap?['close'] ?? true) 'close',
+      if (cleanupMethodsMap?['cancel'] ?? true) 'cancel',
+    };
+
+    return MissingCleanupConfig(
+      ignoredTypesCheckers: [
+        for (final _IgnoredTypes(:name, :packageName) in ignoredTypes)
+          if (packageName.startsWith('dart:'))
+            TypeChecker.fromUrl('$packageName#$name')
+          else
+            TypeChecker.fromName(name, packageName: packageName),
+      ],
+      cleanupMethods: cleanupMethods,
+    );
+  }
+
+  final List<TypeChecker> ignoredTypesCheckers;
+  final Set<String> cleanupMethods;
+}
+
+/// Checks for proper disposal of resources in StatefulWidget classes.
+/// Warns when disposable resources are not properly disposed in the dispose() method.
+class MissingCleanup extends DartLintRule {
+  const MissingCleanup({required this.config})
+    : super(
+        code: const LintCode(
+          name: ruleName,
+          problemMessage:
+              'Resource should be disposed in the dispose() method.',
+          correctionMessage: 'Add disposal of this resource.',
+          errorSeverity: ErrorSeverity.WARNING,
+        ),
+      );
+
+  factory MissingCleanup.fromConfigs(CustomLintConfigs configs) {
+    final config = MissingCleanupConfig.fromConfig(
+      configs.rules[ruleName]?.json.cast<String, YamlNode?>() ?? {},
+    );
+
+    return MissingCleanup(config: config);
+  }
+
+  final MissingCleanupConfig config;
+
+  static const ruleName = 'missing_cleanup';
+
+  @override
+  List<Fix> getFixes() => [_AddDisposeMethod()];
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ErrorReporter reporter,
+    CustomLintContext context,
+  ) {
+    context.registry.addFieldDeclaration((node) {
+      final type = _getFieldDeclarationType(node);
+
+      if (type == null || _isIgnoredInstance(type)) {
+        return;
+      }
+
+      final classNode = _getContainingClass(node);
+      if (classNode == null) {
+        return;
+      }
+
+      if (_isWidgetClass(classNode) &&
+          !_isFieldUsedByConstructor(classNode, node)) {
+        reporter.atNode(node, code);
+        return;
+      }
```

Comment:
How does this check work? Why it matters if something is used in the constructor?